arbnode/sequencer.go

```diff
@@ -114,15 +118,87 @@ func (i *txQueueItem) returnResult(err error) {
 	close(i.resultChan)
 }
 
+type nonceCache struct {
+	cache *containers.LruCache[common.Address, uint64]
+	block common.Hash
+	dirty *types.Header
+}
+
+func newNonceCache(size int) *nonceCache {
+	return &nonceCache{
+		cache: containers.NewLruCache[common.Address, uint64](size),
+		block: common.Hash{},
+		dirty: nil,
+	}
+}
+
+func (c *nonceCache) Matches(header *types.Header) bool {
+	if c.dirty != nil {
+		return c.dirty == header
+	} else {
+		return c.block == header.ParentHash
+	}
+}
+
+func (c *nonceCache) Reset(block common.Hash) {
+	c.cache.Clear()
+	c.block = block
+	c.dirty = nil
+}
+
+func (c *nonceCache) BeginNewBlock() {
```

Comment:
We don't have the previous block hash here because we haven't entered transactionstreamer's mutexes yet, but we'll validate the last block hash when we access the cache.

Reply:
ah, got it! thanks